libnavigation-base/src/test/java/com/mapbox/navigation/base/route/NavigationRouteTest.kt

```diff
@@ -3,14 +3,46 @@ package com.mapbox.navigation.base.route
 import com.mapbox.api.directions.v5.models.DirectionsRoute
 import com.mapbox.api.directions.v5.models.DirectionsWaypoint
 import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.bindgen.ExpectedFactory
 import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.internal.NativeRouteParserWrapper
 import com.mapbox.navigation.testing.FileUtils
+import com.mapbox.navigator.RouteInterface
 import io.mockk.every
 import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.unmockkObject
+import org.json.JSONObject
+import org.junit.After
 import org.junit.Assert.assertEquals
+import org.junit.Before
 import org.junit.Test
 
 class NavigationRouteTest {
+
+    @Before
+    fun setup() {
+        mockkObject(NativeRouteParserWrapper)
+        every {
+            NativeRouteParserWrapper.parseDirectionsResponse(any(), any())
+        } answers {
+            val routesCount = JSONObject(this.firstArg<String>())
+                .getJSONArray("routes")
+                .length()
+            val nativeRoutes = mutableListOf<RouteInterface>().apply {
+                repeat(routesCount) {
+                    add(mockk())
+                }
+            }
+            ExpectedFactory.createValue(nativeRoutes)
+        }
+    }
+
+    @After
+    fun tearDown() {
+        unmockkObject(NativeRouteParserWrapper)
+    }
```

Comment:
This setup is duplicated across various unit tests and I'd love to have a re-usable piece of code for it, unfortunately, we'd need to create a whole new library to place it in. If placed in `libtesting-utils`, it'd create a circular dependency with `libnavigation-base`.

We'd need a new testing library that integrates just `libnavigation-base`.